framework/src/positions/ElementNodePositions.C

```diff
@@ -0,0 +1,88 @@
+//* This file is part of the MOOSE framework
+//* https://mooseframework.inl.gov
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ElementNodePositions.h"
+
+#include "libmesh/parallel_algebra.h"
+
+registerMooseObject("MooseApp", ElementNodePositions);
+
+InputParameters
+ElementNodePositions::validParams()
+{
+  InputParameters params = Positions::validParams();
+  params.addClassDescription("Positions of element nodes.");
+  params += BlockRestrictable::validParams();
+
+  // Element nodes must be sorted to remove duplicates.
+  params.set<bool>("auto_sort") = true;
+  // Gathered locally, should be broadcast on every process
+  params.set<bool>("auto_broadcast") = true;
+
+  return params;
+}
+
+ElementNodePositions::ElementNodePositions(const InputParameters & parameters)
+  : Positions(parameters), BlockRestrictable(this), _mesh(_fe_problem.mesh())
+{
+  // Mesh is ready at construction
+  initialize();
+  // Trigger synchronization as the initialization is distributed
+  finalize();
+}
+
+void
+ElementNodePositions::initialize()
+{
+  clearPositions();
+
+  for (const auto & [node_id, elems_ids] : _fe_problem.mesh().nodeToElemMap())
+  {
+    if (!_mesh.queryNodePtr(node_id))
+      continue;
+
+    // Check the elements associated with the node to see if they're in the block
+    // we're block-restricting. If so, add the node to the positions vector and move
+    // on to the next node (to minimize duplicates).
+    for (const auto & elem_id : elems_ids)
```

Comment:
```suggestion
    for (const auto elem_id : elems_ids)
```
references to trivially copyable members are actually slower

Reply:
Ah, right. I tend to go on autopilot (no pun intended) and throw the `&` in without thinking.